internal/rbd/rbd_util.go

```diff
@@ -478,6 +479,44 @@ func (rv *rbdVolume) flattenRbdImage(ctx context.Context, cr *util.Credentials,
 	return nil
 }
 
+func (rv *rbdVolume) getParentName() (string, error) {
+	rbdImage, err := rv.open()
+	if err != nil {
+		return "", err
+	}
+	defer rbdImage.Close()
+
+	parentPool := make([]byte, 128)
+	parentName := make([]byte, 128)
+	parentSnapname := make([]byte, 128)
+
+	err = rbdImage.GetParentInfo(parentPool, parentName, parentSnapname)
+	if err != nil {
+		return "", err
+	}
+	return string(parentName), nil
+}
+
+func (rv *rbdVolume) flatten() error {
+	rbdImage, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer rbdImage.Close()
+
+	err = rbdImage.Flatten()
+	if err != nil {
+		parent, pErr := rv.getParentName()
```

Comment:
This looks good, although some comments would be in order to make it easier to understand the intended "flatten() returns no error if the image does not have a parent".

Reply:
added a comment.